src/System.Net.Sockets/tests/FunctionalTests/SendFile.cs

```diff
@@ -180,6 +180,100 @@ public void SendFile_Synchronous(IPAddress listenAt, bool sendPreAndPostBuffers,
             File.Delete(filename);
         }
 
+        [Fact]
+        public async Task SyncSendFileGetsCanceledByDispose()
+        {
+            // We try this a couple of times to deal with a timing race: if the Dispose happens
+            // before the operation is started, the peer won't see a ConnectionReset SocketException and we won't
+            // see a SocketException either.
+
+            SocketError? peerSocketError = null;
+            SocketError? localSocketError = null;
+            for (int i = 0; i < 10 && (!peerSocketError.HasValue || !localSocketError.HasValue); i++)
+            {
+                (Socket socket1, Socket socket2) = CreateConnectedSocketPair();
+                using (socket2)
+                {
+                    Task socketOperation = Task.Factory.StartNew(() =>
+                    {
+                        // Create a large file that will cause SendFile to block until the peer starts reading.
+                        string filename = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
```

Comment:
Can we use FileCleanupTestBase instead?  We should be able to put it at the bottom of the class hierarchy.